services/motion/builtin/move_request.go

```diff
@@ -257,6 +301,13 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 				return state.ExecuteResponse{}, err
 			}
 
+			mr.logger.CDebugf(ctx, "CheckPlan inputs: \n currentPosition: %v\n currentInputs: %v\n errorState: %v\n worldstate: %s",
+				spatialmath.PoseToProtobuf(currentPosition.Pose()),
+				currentInputs,
+				spatialmath.PoseToProtobuf(errorState),
+				worldState.String(),
+			)
+
 			if err := motionplan.CheckPlan(
```

Comment:
I chose to keep this as such since we do not know how many cameras a user employed for obstacle detection.

If were to move `CheckPlan` outside the loop entirely _and_ if getting a detection from a singular camera is non-trivial, then as the number of cameras increases, the validity of originally observed detections is likely no longer accurate.
Further, I believe that once we exit the `getTransientDetections` loop, the union of all obstacles would no longer resemble the environment the robot finds itself in.

If still want to take the union of all detections across all cameras, I think we should stick to running `CheckPlan` once per vision service and not for all vision services and all cameras.
I also think that if we are to do so, then we should be running the `getTransientDetections` loop in parallel.

Reply:
This makes sense.